x/config/doc.go

```diff
@@ -0,0 +1,186 @@
+//
+// 	loader := config.NewLoader()
+// 	http.RegisterTransport(loader)
+// 	redis.RegisterTransport(loader)
+//
+// 	builder, err := loader.LoadYAML(yamlConfig)
+// 	if err != nil {
+// 		log.Fatal(err)
+// 	}
+//
+// 	dispatcher, err := builder.BuildDispatcher()
+// 	if err != nil {
+// 		log.Fatal(err)
+// 	}
+//
+// Configuration parameters for the different transports, inbounds, and
+// outbounds are defined in the types fed into the TransportSpec that was
+// registered against the Loader.
+//
+// Configuration
+//
+// The configuration may be specified in YAML or as any Go-level
+// map[string]interface{}. The examples below use YAML for illustration
+// purposes.
+//
+// The configuration accepts the following top-level attributes: name,
+// transports, inbounds, and outbounds.
+//
+// 	name: myservice
+// 	inbounds:
+// 	  # ...
+// 	outbounds:
+// 	  # ...
+// 	transports:
+// 	  # ...
+//
+// Where name specifies the name of the current service. See the following
+// sections for details on the transports, inbounds, and outbounds keys in the
+// configuration.
+//
+// Inbound Configuration
+//
+// The 'inbounds' attribute configures the different ways in which the service
+// receives requests. It is represented as a mapping between inbound transport
+// type and its configuration. For example, the following states that we want
+// to receive requests over HTTP and Redis.
+//
+// 	inbounds:
+// 	  redis:
+// 	    # ...
+// 	  http:
+// 	    # ...
+//
+// (For details on the configuration parameters of individual transport types,
+// check the documentation for the corresponding transport package.)
+//
+// If you want multiple inbounds of the same type, specify a different name
+// for it and add a 'type' attribute to its configuration:
+//
+// 	inbounds:
+// 	  http:
+// 	    # ...
+// 	  http-deprecated:
+// 	    type: http
+// 	    # ...
+//
+// Any inbound can be disabled by adding a 'disabled' attribute.
+//
+// 	inbounds:
+// 	  http:
+// 	    # ...
+// 	  http-deprecated:
+// 	    type: http
+// 	    disabled: true
+// 	    # ...
+//
+// Outbound Configuration
+//
+// The 'outbounds' attribute configures how this service makes requests to
+// other YARPC-compatible services. It is represented as mapping between
+// service name and outbound configuration.
+//
+// 	outbounds:
+// 	  keyvalue:
+// 	    # ..
+// 	  anotherservice:
+// 	    # ..
+//
+// (For details on the configuration parameters of individual transport types,
+// check the documentation for the corresponding transport package.)
+//
+// The outbound configuration for a service has at least one of the following
+// keys: unary, oneway. These specify the configurations for the corresponding
+// RPC types for that service. For example, the following specifies that we
+// make Unary requests to keyvalue service over HTTP and Oneway requests over
+// Redis.
+//
+// 	keyvalue:
+// 	  unary:
+// 	    http:
+// 	      # ...
+// 	  oneway:
+// 	    redis:
```

Comment:
I reeeeeeeeeeeeeeeeaaaaaaally hope no one wants to do this.  it sort of muddles what an "outbound" is, because I could have a configuration that points at two completely different services but use the same "key" for the service...  *shrug*